### PR TITLE
Currently, when get timeout value from testrail.cfg or from command l…

### PR DESCRIPTION
**Pull request by cuongld2**
- Currently, when get timeout value from testrail.cfg or from command line, pytest_testrail get it as string.
Need to convert to float value in order to run
- Modify pytest version in requirements base.txt to latest pytest version

### CODE REVIEW

**Comment by cuongld2** [comment]
Dear @allankp @PeterNgTr ,
I refactored and added unit test.
Please take a look.
Thank you!

File: pytest_testrail/testrail_api.py [hunk 1]
```diff
@@ -69,6 +74,8 @@ def send_get(self, uri, **kwargs):
         cert_check = kwargs.get('cert_check', self.cert_check)
         headers = kwargs.get('headers', self.headers)
         timeout = kwargs.get('timeout', self.timeout)
+        # Convert timeout from string to float if defined in testrail config file or from command line
+        timeout = convert_to_float(timeout)
```
Comment (allankp) [inline pytest_testrail/testrail_api.py]: `isinstance(timeout, float) if False else float(timeout)`

This is much cleaner and doesn't require the method above, sorry I should have spotted this before. Also its a bit unclear why we get a timeout in each method. Seems off to require a different timeout for get / send. Not your problem but i'll look at it later.

File: pytest_testrail/testrail_api.py [hunk 2]
```diff
@@ -108,6 +115,8 @@ def send_post(self, uri, data, **kwargs):
...
```
Comment (allankp) [inline pytest_testrail/testrail_api.py]: As Per comment above

File: pytest_testrail/testrail_api.py (hunk 1 above)
Comment (cuongld2) [reply pytest_testrail/testrail_api.py]: Dear @allankp ,
Thanks for the comment.
isinstance(timeout, float) if False else float(timeout)
I think you're missing an assignment and does not check if timeout is None.
My current implementation is like follow:
        if self.timeout is not None:
            self.timeout = isinstance(self.timeout, float) if False else float(self.timeout)

Also, I just removed the timeout we get for send_post and send_get.
Please take a look at what changes.
Cheers!

File: tests/test_plugin.py [hunk 3]
```diff
@@ -365,3 +364,19 @@ def test_skip_missing_correlation_tests(api_client, pytest_test_items):
 
     assert not pytest_test_items[0].get_closest_marker('skip')
     assert not pytest_test_items[1].get_closest_marker('skip')
+
+
+def test_check_timeout_convert_string():
```
Comment (allankp) [inline tests/test_plugin.py]: This test is not actually testing the plugin, you need to used the `api_client` fixture.

File: pytest_testrail/testrail_api.py (hunk 1 above)
Comment (allankp) [reply pytest_testrail/testrail_api.py]: Fair enough I missed that, looks cleaner now. thanks.

File: tests/test_plugin.py (hunk 3 above)
Comment (cuongld2) [reply tests/test_plugin.py]: Dear @allankp ,
Thanks for your comment.
I've just modified the unit test.
Please have a look.
Peace!!!

**Pull request merged**